### PR TITLE
ISSUE-330 # Send header kafka producer

### PR DESCRIPTION
**Pull request by kt09**
# <Feature Title>

PR Branch
**_#ADD LINK TO THE PR BRANCH_**

## Motivation and Context
@authorjapps -- Please review and approve. This contribution is not tested.

## Checklist:

* [ ] Unit tests added?

* [ ] Integration tests added?

* [ ] Test names are meaningful?

* [ ] Feature manually tested?

* [ ] Branch build passed?

* [ ] No 'package.*' in the imports

* [ ] Relevant Wiki page updated with clear instruction for the end user

* [ ] Http test added to `http-testing` module(if applicable) ?

* [ ] Kafka test added to `kafka-testing` module(if applicable) ?

### CODE REVIEW

File: core/src/main/java/org/jsmart/zerocode/core/kafka/helper/KafkaProducerHelper.java [hunk 1]
```diff
@@ -53,25 +53,53 @@ public static void validateProduceRecord(List producerRecords) {
...
+        // constructor without header
+        if (recordToSend.headers() == null){
+            return new ProducerRecord(topicName,
+                    recordToSend.partition(),
+                    recordToSend.timestamp(),
+                    recordToSend.key(),
+                    recordToSend.value());
+        }else {
+            // constructor with header
+            return new ProducerRecord(topicName,
```
Comment (authorjapps) [inline core/src/main/java/org/jsmart/zerocode/core/kafka/helper/KafkaProducerHelper.java:65]: Looks great!
Can you cove these with some tests?

File: core/src/main/java/org/jsmart/zerocode/core/kafka/helper/KafkaProducerHelper.java [hunk 2]
```diff
@@ -53,25 +53,53 @@ public static void validateProduceRecord(List producerRecords) {
...
+                    recordToSend.getKey(),
+                    // --------------------------------------------
+                    // It's a JSON as String. Nothing to worry !
+                    // Kafka StringSerializer needs in this format.
+                    // --------------------------------------------
+                    recordToSend.getValue().toString()
+            );
+        }else{
+            // constructor with header
+            return new ProducerRecord(topicName,
```
Comment (authorjapps) [inline core/src/main/java/org/jsmart/zerocode/core/kafka/helper/KafkaProducerHelper.java:90]: 👍 
Can you cover this flow with tests too?
Please check the PR guidelines!

File: core/src/main/java/org/jsmart/zerocode/core/kafka/send/message/ProducerJsonRecord.java [hunk 3]
```diff
@@ -3,27 +3,38 @@
...
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.kafka.common.header.Headers;
 
 // TODO - add timestamp, partition key etc
 public class ProducerJsonRecord<K> {
     private final K key;
     private final JsonNode jsonKey;
     private final JsonNode value;
+    private final Headers headers;
```
Comment (authorjapps) [inline core/src/main/java/org/jsmart/zerocode/core/kafka/send/message/ProducerJsonRecord.java:13]: Please cover with unit tests.
> Just unit tests are enough for this POJO

**Comment by nirmalchandra** [comment]
PR is Not relevant any more, as the issue has been resolved by another PR.

**Pull request closed without merging**